presto-main/src/main/java/com/facebook/presto/execution/SqlTask.java

```diff
@@ -429,10 +436,20 @@ public TaskInfo updateTask(
         return getTaskInfo();
     }
 
-    public ListenableFuture<BufferResult> getTaskResults(OutputBufferId bufferId, long startingSequenceId, DataSize maxSize)
+    public ListenableFuture<BufferResult> getTaskResults(OutputBufferId bufferId, long startingSequenceId, DataSize maxSize, String communicationSlug)
     {
         requireNonNull(bufferId, "bufferId is null");
         checkArgument(maxSize.toBytes() > 0, "maxSize must be at least 1 byte");
+        requireNonNull(communicationSlug);
+
+        String currentCommunicationSlug = this.communicationSlug.get();
+        // Take care of situation where in machine has not received the first updateTask from the coordinator
+        if (currentCommunicationSlug == null) {
+            return Futures.immediateFuture(BufferResult.emptyResults(taskInstanceId, startingSequenceId, false));
```

Comment:
Probably this should be `requireNotNull`? We will always call `updateTask` with a communicationSlug.